modules/javafx.graphics/src/main/native-glass/ios/GlassViewDelegate.m

```diff
@@ -375,6 +375,19 @@ - (void)handleRotateGesture:(UIRotationGestureRecognizer*)sender {
 }
 
 
+- (void)handleLongPressGesture:(UILongPressGestureRecognizer*)sender {
+    if (sender.state == UIGestureRecognizerStateBegan) {
+        // Simulate right-click
+        CGPoint viewPoint = [sender locationInView:self.uiView.superview];
+        [self sendJavaMouseEvent:viewPoint type:com_sun_glass_events_MouseEvent_ENTER button:com_sun_glass_events_MouseEvent_BUTTON_NONE];
+        [self sendJavaMouseEvent:viewPoint type:com_sun_glass_events_MouseEvent_DOWN button:com_sun_glass_events_MouseEvent_BUTTON_RIGHT];
+    } else if (sender.state == UIGestureRecognizerStateEnded) {
+        // Prevent touch ended event
```

Comment:
The "right-click" event (enter+right down) is simulated when the long press event is detected (`UIGestureRecognizerStateBegan `, which happens after 0.5 seconds by default). This will trigger the ContextMenu.

However, the user is still pressing for a few more time. After releasing the tap, (`UIGestureRecognizerStateEnded`), there is no need of adding a release event (exit+right up), as this will probably will close the ContextMenu. The user now can select any of the options from the menu, or simply tap elsewhere to close it.